### ISSUE TITLE
[ok.ru] cannot get all quality of video

### ISSUE THREAD

**Issue by jazz1611** [yt-dlp/yt-dlp#4462]
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.07.18** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Provide a description that is worded well enough to be understood

hello,
i can see many quality video in browser but yt-dlp can get only hd or sd in format list.
https://ok.ru/video/1510273649159
https://ok.ru/video/1951798069873

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[root@CentOS-79-64-minimal ~]# youtube-dl -UvF "https://ok.ru/video/1951798069873"
[debug] Command-line config: ['-UvF', 'https://ok.ru/video/1951798069873']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out UTF-8, error UTF-8, screen UTF-8
[debug] yt-dlp version 2022.07.18 [135f05ef6]
[debug] Lazy loading extractors is disabled
[debug] Python 3.7.13 (CPython 64bit) - Linux-3.10.0-1160.59.1.el7.x86_64-x86_64-with-centos-7.9.2009-Core (glibc 2.17)
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.0.1 (fdk,setts), ffprobe 5.0.1, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.07.18, Current version: 2022.07.18
yt-dlp is up to date (2022.07.18)
[debug] [Odnoklassniki] Extracting URL: https://ok.ru/video/1951798069873
[Odnoklassniki] 1951798069873: Downloading desktop webpage
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 1951798069873:
ID EXT RESOLUTION │ PROTO │ VCODEC  ACODEC
───────────────────────────────────────────
hd mp4 unknown    │ https │ unknown unknown

[root@CentOS-79-64-minimal ~]# youtube-dl -UvF "https://ok.ru/video/1510273649159"
[debug] Command-line config: ['-UvF', 'https://ok.ru/video/1510273649159']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out UTF-8, error UTF-8, screen UTF-8
[debug] yt-dlp version 2022.07.18 [135f05ef6]
[debug] Lazy loading extractors is disabled
[debug] Python 3.7.13 (CPython 64bit) - Linux-3.10.0-1160.59.1.el7.x86_64-x86_64-with-centos-7.9.2009-Core (glibc 2.17)
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.0.1 (fdk,setts), ffprobe 5.0.1, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.07.18, Current version: 2022.07.18
yt-dlp is up to date (2022.07.18)
[debug] [Odnoklassniki] Extracting URL: https://ok.ru/video/1510273649159
[Odnoklassniki] 1510273649159: Downloading desktop webpage
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 1510273649159:
ID EXT RESOLUTION │ PROTO │ VCODEC  ACODEC
───────────────────────────────────────────
sd mp4 unknown    │ https │ unknown unknown
```

**Comment by dirkf** [comment]
In addition to the m3u8 manifest URL expected at `metadata['hlsManifestUrl']`, the extractor should also try `metadata['ondemandHls']`. In addition to the embedded DASH manifest expected at `metadata['metadataEmbedded']`, the extractor should also try for a DASH manifest URL at `metadata['ondemandDash']`.

**Comment by jazz1611** [comment]
hello @pukkandan ,

you can try with this video https://ok.ru/video/4588804246201

```
C:\Users\admin>yt-dlp -UvF "https://ok.ru/video/4588804246201"
[debug] Command-line config: ['-UvF', 'https://ok.ru/video/4588804246201']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.09.01 [5d7c7d6] (win32_exe)
[debug] Python 3.8.10 (CPython 64bit) - Windows-10-10.0.22000-SP0
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.1-full_build-www.gyan.dev (setts), ffprobe 5.1-full_build-www.gyan.dev, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Loaded 1670 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.09.01, Current version: 2022.09.01
yt-dlp is up to date (2022.09.01)
[debug] [Odnoklassniki] Extracting URL: https://ok.ru/video/4588804246201
[Odnoklassniki] 4588804246201: Downloading desktop webpage
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 4588804246201:
ID   EXT RESOLUTION │ PROTO │ VCODEC  ACODEC
─────────────────────────────────────────────
full mp4 unknown    │ https │ unknown unknown
```

![image](https://user-images.githubusercontent.com/1846222/188961882-c928bbca-6fa3-43fa-84bd-153b5652f6e8.png)

**Comment by bashonly** [comment]
Patch available: https://github.com/yt-dlp/yt-dlp/issues/6442#issuecomment-1455181924

and if anyone wants to PR that, include this patch too: https://github.com/yt-dlp/yt-dlp/issues/6442#issuecomment-1455181924
